test/e2e/suites/e2e/e2e_test.go

```diff
@@ -569,33 +570,61 @@ var _ = Describe("e2e tests [PR-Blocking]", func() {
 					Equal(*machine.Spec.FailureDomain),
 					fmt.Sprintf("Server %s was not scheduled in the correct AZ", machine.Name))
 
-				// Check that all machines have the expected boot volume
+				// Check that all machines have the expected volumes:
+				// - 1 boot volume
+				// - 1 additional volume
 				volumes := server.AttachedVolumes
-				Expect(volumes).To(HaveLen(1))
+				Expect(volumes).To(HaveLen(2))
 
-				bootVolume, err := shared.GetOpenStackVolume(e2eCtx, volumes[0].ID)
+				firstVolume, err := shared.GetOpenStackVolume(e2eCtx, volumes[0].ID)
 				Expect(err).NotTo(HaveOccurred(), "failed to get OpenStack volume %s for machine %s", volumes[0].ID, machine.Name)
+				secondVolume, err := shared.GetOpenStackVolume(e2eCtx, volumes[1].ID)
+				Expect(err).NotTo(HaveOccurred(), "failed to get OpenStack volume %s for machine %s", volumes[0].ID, machine.Name)
+
+				bootVolume := firstVolume
+				additionalVolume := secondVolume
+				// The boot volume is the one with the "bootable" flag set.
+				if firstVolume.Bootable != "true" { // This is genuinely a string, not a bool
```

Comment:
I need to do more testing to confirm that listing volumes provide an output ordered by creation time. I'll look at that today because I agree that this is ugly :smile: .

Reply:
I tested it and indeed it seems that the volume used for boot is always first.